<launch>
    
    <!-- Set the joystick input -->
    <arg name="joy_config" default="xbox" />
    <arg name="config_filepath" default="$(find joystick_control)/config/$(arg joy_config).config.yaml" />
    <arg name="joystick" default="js0"/>

    <arg name="base_topic" default="cmd_vel"/>
    <arg name="camera_topic" default="pitch_yaw"/>


    <!-- Base and Main camera control-->
    <!-- joy node -->
    <node respawn="true" pkg="joy" type="joy_node" name="teleop_joystick" >
        <param name="dev" type="string" value="/dev/input/$(arg joystick)" />
        <param name="deadzone" value="0.12" />
        <param name="autorepeat_rate" value="20" />
    </node>

    <!-- teleoperation node -->
    <node name="joystick_control_node" pkg="joystick_control" type="joystick_control_node" output="screen"> 
        <!-- axes -->
        <rosparam command="load" file="$(arg config_filepath)" />

        <!-- topics -->
        <remap from="cmd_vel" to="$(arg base_topic)" />
        <remap from="pitch_yaw" to="$(arg camera_topic)" />
    </node> 

    <node name="joystic_control_node_1" pkg="joystick_control" type="joystick_control.py" output="screen">
        <remap from="cmd_vel" to="$(arg base_topic)" />
        <remap from="pitch_yaw" to="$(arg camera_topic)" />
    </node>
    <param name="device" value="/dev/input/event9" />
    <node name="joy_feedback_ros1" pkg="joy_feedback_ros" type="joy_feedback" output="screen">
    </node>

</launch>
